### PR TITLE
Community Link

### PR DESCRIPTION
**Pull request by rayyank10**
added community link component

Resolves: https://jira.corp.adobe.com/browse/MWPW-147082

Screen Shot 2024-05-16 at 7 30 05 PM
Test URLs:

- Before: https://main--ecc-milo--adobecom.hlx.page/
- After: https://commLink--ecc-milo--adobecom.hlx.page/

### CODE REVIEW

File: blocks/event-community-link-component/event-community-link-component.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  await Promise.all([
+    import(`${miloLibs}/deps/lit-all.min.js`),
+    import(`${miloLibs}/features/spectrum-web-components/dist/textfield.js`),
+    import(`${miloLibs}/features/spectrum-web-components/dist/checkbox.js`),
+  ]);
+
+  el.classList.add('form-component');
+  generateToolTip(el);
+  const rows = [...el.querySelectorAll(':scope > div')];
```
Comment (qiyundai) [inline blocks/event-community-link-component/event-community-link-component.js:64]: Have you tried without the array spread? I think JavaScript nodeList lets you access node by index.

File: blocks/event-community-link-component/event-community-link-component.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const cols = row.querySelectorAll(':scope > div');
+  if (!cols.length) return null;
+  const [checkboxText, placeholderCol] = cols;
+  const text = placeholderCol.textContent.trim();
+
+  let input;
+
+  if (type === 'text') {
+    input = createTag('sp-textfield', {
+      id: 'community-URL-details',
```
Comment (qiyundai) [inline blocks/event-community-link-component/event-community-link-component.js:17]: Nit: this stood out to me a little as I don't see many uppercase letters in id/class across Franklin projects but it's totally up to you.

**Review by qiyundai — COMMENTED** [review]
Looking good! Just spotted a few non-blocking things. I'll check it visually on my localhost tomorrow and merge it with or without these few things addressed.

File: blocks/event-community-link-component/event-community-link-component.js (hunk 1 above)
Comment (rayyank10) [reply blocks/event-community-link-component/event-community-link-component.js:64]: I used the spread intentionally  but its no worries if you prefer me to change it

File: blocks/event-community-link-component/event-community-link-component.js (hunk 2 above)
Comment (rayyank10) [reply blocks/event-community-link-component/event-community-link-component.js:17]: I can fix this for consistency

**Pull request merged**